wallet-connect/src/client.rs

```diff
@@ -58,9 +59,9 @@ impl Default for ClientChannelMessage {
 /// (holds the middleware trait implementations for ethers)
 /// FIXME: a way for persisting and recovering the client state
 /// (a callback function or a path which the client can control?)
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct Client {
-    connection: Connector,
+    connection: Arc<Mutex<Connector>>,
     // to make receive channel valid, sender channel should be open
     callback_channel: Option<UnboundedSender<ClientChannelMessage>>,
```

Comment:
one issue with clone may be for the callback that the receiver may be sent the same message multiple times?
maybe good to document that in the description or if it's an issue, have a custom Clone implementation (instead of the auto-derive one) that would always set `callback_channel` as None

Reply:
ok